src/engraving/rendering/dev/tlayout.cpp

```diff
@@ -2626,9 +2626,22 @@ static void layoutGlissando(const Glissando* item, LayoutContext& ctx, Glissando
             return false;
         }
     });
-    offs1.rx() += cr1shape.right() - anchor1->pos().x();
+
+    double yAbove = anchor1->ldata()->pos().y() + anchor1->ldata()->bbox().topRight().y();
+    double yBelow = yAbove + anchor1->ldata()->bbox().height();
+    offs1.rx() += cr1shape.rightMostEdgeAtHeight(yAbove, yBelow) - anchor1->pos().x();
     if (!cr2->staff()->isTabStaff(cr2->tick())) {
-        offs2.rx() -= cr2->shape().left() + anchor2->pos().x();
+        double yAbove2 = anchor2->ldata()->pos().y() + anchor2->ldata()->bbox().topLeft().y();
+        double yBelow2 = yAbove2 + anchor2->ldata()->bbox().height();
+        double noteMiddle = yAbove2 + anchor2->ldata()->bbox().height() / 2;
+        if (upDown != 0) {
+            // Only check top/bottom half of note depending on gliss approach direction
+            // to avoid clearing acidentals the line won't collide with
+            yAbove2 = upDown == 1 ? noteMiddle - 1 : yAbove2;
```

Comment:
Is it intentional that there is an "absolute" `- 1` here? Shouldn't it be relative to spatium or line distance? (Or _is_ it that already and am I missing that?)

Reply:
Apologies - this was to make sure the ledger line was included in the left most edge check.  I've replaced it with the thickness of the ledger line